plutus-core/cost-model/create-cost-model/CostModelCreation.hs

```diff
@@ -254,10 +342,64 @@ lessThanEqualsInteger cpuModelR = do
   cpuModel <- readModelMinSize cpuModelR
   pure $ CostingFun (ModelTwoArgumentsMinSize cpuModel) boolMemModel
 
+
+---------------- Bytestrings ----------------
+
+appendByteString :: MonadR m => (SomeSEXP (Region m)) -> m (CostingFun ModelTwoArguments)
+appendByteString cpuModelR = do
+  cpuModel <- readModelAddedSizes cpuModelR
+  -- The buffer gets reallocated
+  let memModel = ModelTwoArgumentsAddedSizes $ ModelAddedSizes 0 1
+  pure $ CostingFun (ModelTwoArgumentsAddedSizes cpuModel) memModel
+
+
+-- The things marked with ### comments below are plausible guesses at the form
+-- of the model to get it fixed for the HF.  Later we'll have to add benchmarks
+-- to get data and R code to fit models for all of them.
+
+-- ### TODO: get model from R ###
+consByteString ::  MonadR m => (SomeSEXP (Region m)) -> m (CostingFun ModelTwoArguments)
+consByteString _ = do
+      let cpuModel = ModelAddedSizes 10000 1000  -- 10000 + 1000*(x+y)
```

Comment:
Shouldn't this be linear in `x` and `y`? Which isn't quite the same as being linear in `x+y`? And indeed maybe it should only depend on the size of the bytestring?

Reply:
I was probably thinking about `appendByteString` there.  I'll look at that again.

Reply:
That gave me a chance to use one of the unused costing function types, so being extravagant paid off.